war/src/test/java/io/lumeer/engine/controller/UserRoleFacadeIntegrationTest.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * -----------------------------------------------------------------------\
+ * Lumeer
+ *  
+ * Copyright (C) 2016 - 2017 the original author or authors.
+ *  
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -----------------------------------------------------------------------/
+ */
+package io.lumeer.engine.controller;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.lumeer.engine.IntegrationTestBase;
+import io.lumeer.engine.annotation.SystemDataStorage;
+import io.lumeer.engine.api.LumeerConst;
+import io.lumeer.engine.api.data.DataStorage;
+import io.lumeer.engine.provider.DataStorageProvider;
+
+import org.jboss.arquillian.junit.Arquillian;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import javax.inject.Inject;
+
+/**
+ * Tests for UserRoleFacade
+ */
+@RunWith(Arquillian.class)
+public class UserRoleFacadeIntegrationTest extends IntegrationTestBase {
+
+   @Inject
+   @SystemDataStorage
+   private DataStorage systemDataStorage;
+
+   @Inject
+   private DataStorageProvider dataStorageProvider;
+
+   @Inject
+   private UserRoleFacade userRoleFacade;
+
+   @Before
+   public void setUp() throws Exception {
+      systemDataStorage.dropManyDocuments(LumeerConst.Project.UserRoles.COLLECTION_NAME, "{}");
+   }
+
+   @Test
+   public void projectUserRolesTest() throws Exception {
+
+      final String organization = "oganization1";
+      final String project1 = "project1";
+      final String project2 = "project2";
+
+      final String userRole1 = "userRole1";
+      final String userRole2 = "userRole2";
+      final String userRole3 = "userRole3";
+
+      final String coreRole1 = "coreRole1";
+      final String coreRole2 = "coreRole2";
+      final String coreRole3 = "coreRole3";
+
+      Map<String, List<String>> rolesProject1 = userRoleFacade.readRoles(organization, project1);
+      Map<String, List<String>> rolesProject2 = userRoleFacade.readRoles(organization, project2);
+      assertThat(rolesProject1).isEmpty();
+      assertThat(rolesProject2).isEmpty();
+
+      // create test
```

Comment:
You do not need to spend time editing this now but in the future try not to write such long test methods if they can be split into multiple smaller ones. It is then easier to find where the problem is when a test starts failing.

The tests can be split according to the comments you put there.